src/common/low_precision_transformations/src/multiply.cpp

```diff
@@ -63,91 +63,91 @@ bool MultiplyTransformation::transform(TransformationContext& context, ov::pass:
+            new_scales_values :
+            dequantization2.subtract == nullptr ?
+                dequantization2.data :
+                NetworkHelper::optimizeSubtract(dequantization2.subtract);
+
+        auto const new_multiply = (in1.get_element_type() == multiply->get_output_element_type(0)) &&
+                                  (in2.get_element_type() == multiply->get_output_element_type(0)) ?
+            std::make_shared<ov::opset1::Multiply>(in1, in2) :
+            std::make_shared<ov::op::TypeRelaxed<ov::opset1::Multiply>>(
+                std::vector<ov::element::Type>{ deqPrecision, deqPrecision },
+                std::vector<ov::element::Type>{ multiply->get_output_element_type(0) },
+                ov::op::TemporaryReplaceOutputType(in1, deqPrecision).get(),
+                ov::op::TemporaryReplaceOutputType(in2, deqPrecision).get());
+
+        replace_node(multiply, new_multiply);
+        updateOutput(context, new_multiply, multiply);
+
+        return true;
     }
 
-    replace_node(multiply, newMultiply);
-    updateOutput(context, newMultiply, multiply);
-
-    if (fullPathIndex != -1) {
-        NetworkHelper::foldDequantization(newMultiply, fullPathIndex, defaultPrecisions);
+    auto new_scales_values = fold<ov::opset1::Multiply>(dequantization1.multiplyConstant, dequantization2.multiplyConstant);
+    if (!ov::is_type<ov::opset1::Constant>(new_scales_values)) {
+        return false;
     }
 
+    const Output<Node> in1 = dequantization1.subtract == nullptr ?
+        dequantization1.data :
+        NetworkHelper::optimizeSubtract(dequantization1.subtract);
+
+    const Output<Node> in2 = dequantization2.subtract == nullptr ?
+        dequantization2.data :
+        NetworkHelper::optimizeSubtract(dequantization2.subtract);
+
+    // in1 & in2 can have different input types
+    const auto new_multiply = (in1.get_element_type() == deqPrecision) &&
+                              (in2.get_element_type() == deqPrecision) ?
+        std::make_shared<ov::opset1::Multiply>(in1, in2) :
+        std::make_shared<ov::op::TypeRelaxed<ov::opset1::Multiply>>(
+            std::vector<ov::element::Type>{ deqPrecision, deqPrecision },
+            std::vector<ov::element::Type>{ deqPrecision },
+            ov::op::TemporaryReplaceOutputType(in1, deqPrecision).get(),
+            ov::op::TemporaryReplaceOutputType(in2, deqPrecision).get());
+
+    NetworkHelper::copyInfo(multiply, newMultiply);
+
+    auto new_scales = (new_multiply->get_output_element_type(0) == multiply->get_output_element_type(0)) &&
+                      (new_scales_values->get_output_element_type(0) == multiply->get_output_element_type(0)) ?
+        std::make_shared<ov::opset1::Multiply>(new_multiply, new_scales_values) :
+        std::make_shared<ov::op::TypeRelaxed<ov::opset1::Multiply>>(
+            ov::opset1::Multiply(new_multiply, new_scales_values),
+            multiply->get_output_element_type(0));
```

Comment:
If `new_multiply`'s element type is not equal to `new_scales_values`'s element type, we should also override input precisions, right?
BTW do we really have big overheads on `TypeRelaxed` creation in comparison with standard nodes? If not, maybe we can always create `TypeRelaxed` multiply in order to avoid a cumbersome code blocks with ternary operators?

Reply:
Thanks for comment!
1) `new_multiply` & `new_scales_values` output port type is checked by condition:
`
auto new_scales = (new_multiply->get_output_element_type(0) == multiply->get_output_element_type(0)) &&
                              (new_scales_values->get_output_element_type(0) == multiply->get_output_element_type(0)) ?
`
2) If we don't need `TypeRelaxed`, let's dont use it.
When you write `overheads` do you mean performance overhead? I do know: there are cases when we have excess `TypeRelaxed` ops (specially in TypeRelaxed replacer), but it's not ideal by several reasons. One of them is type validation breakage when it's not necessary.

My suggestion: keep as is